src/webapi/node.rs

```diff
@@ -362,6 +362,19 @@ pub struct Node( Reference );
 impl IEventTarget for Node {}
 impl INode for Node {}
 
+impl Node {
+    /// Attempt to create the `Node` from raw html.
+    ///
+    /// Note that this uses a `span` node as the parent.
+    pub fn from_html(html: &str) -> Result<Node, ConversionError> {
+        Node::try_from(js! {
+            var span = document.createElement("span");
+            span.innerHTML = @{html};
+            return span;
+        })
+    }
```

Comment:
1. Automatically creating a `span` here seems kind-of arbitrary to me and potentially confusing, e.g. if I do:

```
Node::from_html( "<div>Hello!</div>" )
```

I'd expect it to return a `<div>`; the fact that it returns a `<span>` is non-obvious from the method signature itself, so *if* we're going to add a method like this then I think it should require that it contains only a single root node and return that it.

2. `innerHTML` this can throw a `SyntaxError` if the `document` is XML (AFAIK this happens when we have an XHTML page), e.g. similar to this:

```
var xmldoc = document.implementation.createDocument(null, null);
var span = xmldoc.createElement("span"); 
span.innerHTML = "<div";
```

Reply:
1. I hadn't considered that idea, but that sounds good.
2. We could use [`DOMParser`](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) and force the user to specify the type: normally `"text/html"`, but can also be `"image/svg+xml"` or `"application/xml"`.

However, I'm perplexed since your code _is_ a syntax error. The folowing works for me:

```
        var xmldoc = document.implementation.createDocument(null, null);
        var span = xmldoc.createElement("span");
        span.innerHTML = "<div></div>";·
```

Reply:
> However, I'm perplexed since your code is a syntax error. The folowing works for me:

That is the point. (: (Although AFAIK the code itself is valid, it's just that a `SyntaxError` is thrown. Unless you meant something else here?)

In my snippet the assignment to `innerHTML` throws a `SyntaxError` exception. That was just an example to illustrate that we need to handle that `SyntaxError` here somehow if we're going to use `innerHTML`.